### PR TITLE
Hydra

### PR DESCRIPTION
**Pull request by XuhuiZhou**

### CODE REVIEW

File: BackTranslation/measure_blue_score.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 
 def measure_bleu_score(args, model, tokenizer, data_source):
-    translation_task = f"translation_{args.source_lang_id}_to_{args.target_lang_id}"
+    translation_task = f"translation_{args.data.source_lang_id}_to_{args.data.data.target_lang_id}"
```
Comment (shanest) [inline BackTranslation/measure_blue_score.py:16]: Why is the second one `args.data.data.target_lang_id` and not just `args.data.target_lang_id`?

File: Configs/backtranslate/bt_baseline.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+defaults:
+  - data: en-de
+  - train_eval: train_eval_baseline
```
Comment (shanest) [inline Configs/backtranslate/bt_baseline.yaml:3]: Small thing, if not too much work: all the values for `train_eval` that are `train_eval_X` could just be `X`, e.g. `- train_eval: baseline`

File: Configs/backtranslate/bt_secondary.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  - _self_
+
+# Path to load model for generation
+model_path: facebook/mbart-large-cc25
+output_dir: ./Output/bt_en-de_initial/
+# lang_pair: de-en # candidates: neen; sien; zh-en
+
+# New root to store output of backtranslation. This can be useful
+# when you are running low of local storage. Used in combination with 
+# an output directory path passed in via config.
```
Comment (shanest) [inline Configs/backtranslate/bt_secondary.yaml:13]: Thanks for the comment!

File: Configs/ec/data/data_captions.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Loading and saving data:
```
Comment (shanest) [inline Configs/ec/data/data_captions.yaml:1]: In this, and similar files, use variables? https://omegaconf.readthedocs.io/en/latest/usage.html#variable-interpolation

For example: 
```
base_dir: ./DataLink/captioning_new_new
train_captions: ${base_dir}/en_captions_train.jsonl
```

File: RunScripts/en-de_ablation_pipeline.sh [hunk 5]
```diff
@@ -35,19 +35,21 @@ python -u -m EC_finetune --config Configs/${EC_CONFIG}.yml \
...
-    --output_dir ${OUTPUT_DIR} \
-    --model_path ${OUTPUT_DIR}/best_bleu
+# cp /projects/unmt/communication-translation/Data/translation_references/de-en.* ${OUTPUT_DIR}
+# python -u BackTranslation/translate.py --config Configs/test_en2de_translate.yml \
+#     --output_dir ${OUTPUT_DIR} \
+#     --model_path ${OUTPUT_DIR}/best_bleu
+# python -u BackTranslation/translate.py --config Configs/test_de2en_translate.yml \
+#     --output_dir ${OUTPUT_DIR} \
+#     --model_path ${OUTPUT_DIR}/best_bleu
+
```
Comment (shanest) [inline RunScripts/en-de_ablation_pipeline.sh:52]: Why is everything commented out?

File: RunScripts/en-de_hydra_pipeline.sh [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    backtranslate.train_eval.seed=2 \
+    backtranslate.model_path=Output/en-de_pipeline/ec_${EX_ABBR} \
+    backtranslate.output_dir=${OUTPUT_DIR}
+
+# python -u BackTranslation/backtranslate.py --config Configs/${BT_SECONDARY_CONFIG}.yml \
+#     --seed_override 2 \
+#     --model_dir_override Output/en-de_pipeline/ec_${EX_ABBR} \
+#     --output_dir_override ${OUTPUT_DIR}
+
+# Do test
```
Comment (shanest) [inline RunScripts/en-de_hydra_pipeline.sh:55]: Can we put testing back in main script, or did we decide to always do testing as a separate script?

**Review by shanest — CHANGES_REQUESTED** [review]
Thanks, Xuhui! Really great stuff!! This will make our lives much much easier :)  Just a few quick comments to look at before merging

File: BackTranslation/measure_blue_score.py (hunk 1 above)
Comment (XuhuiZhou) [reply BackTranslation/measure_blue_score.py:16]: Good catch, I think it should be the latter one. Fixed

File: Configs/backtranslate/bt_baseline.yaml (hunk 2 above)
Comment (XuhuiZhou) [reply Configs/backtranslate/bt_baseline.yaml:3]: not at all, changed

File: RunScripts/en-de_hydra_pipeline.sh (hunk 6 above)
Comment (XuhuiZhou) [reply RunScripts/en-de_hydra_pipeline.sh:55]: I think we've decided to always to testing as a separate script.

File: RunScripts/en-de_ablation_pipeline.sh (hunk 5 above)
Comment (XuhuiZhou) [reply RunScripts/en-de_ablation_pipeline.sh:52]: Because we are not doing the test here.

File: Configs/ec/data/data_captions.yaml (hunk 4 above)
Comment (XuhuiZhou) [reply Configs/ec/data/data_captions.yaml:1]: done!

File: RunScripts/en-de_ablation_pipeline.sh (hunk 5 above)
Comment (shanest) [reply RunScripts/en-de_ablation_pipeline.sh:52]: But this is a lot more than testing that's commented out; pretty much everything is

**Review by shanest — COMMENTED** [review]
I'm going to go ahead and merge this in the interest of time, but please look at `RunScripts/en-de_ablation_pipeline.sh` and my comment there.  Not sure why basically the entire script is commented out, not just testing.  Do we just need new scripts for the hydra version?

**Pull request merged**